### PR TITLE
Added "Shorten Url" command

### PR DESCRIPTION
**Pull request by Olian04**

### CODE REVIEW

**Comment by cliffordfajardo** [comment]
@Olian04 - thanks! I'll take a look at this later today when I'm on my pc 😄

**Comment by Olian04** [comment]
@cliffordfajardo Cool, I'l be heading of to bed so i wont be available for a while.

**Comment by Olian04** [comment]
_(Explanation for the cryptic merge message above...)_

There was a merge conflict regarding two almost identical solutions to https://github.com/cliffordfajardo/cato/issues/8

The usage of the function copyToClipboard, like this: 
```
// PR
copyToClipboard(activeTabUrl, () => {
   window.close();
});
```
instead of explicitly duping code, like this: 
```
// Master
document.addEventListener('copy', (event) => {
  event.preventDefault() //this doesn't work w/o this. Investigate why for exploration's sake?
  const text = activeTabUrl
  event.clipboardData.setData('text/plain', text)
}, {once: true})
ocument.execCommand('copy')
window.close()
```
will work because of commit 3e6531a, where i extracted the above snippet into the function copyToClipboard, and commit 7d72671, where i moved the copyToClipboard function into the `util.js` file.

**Comment by cliffordfajardo** [comment]
@Olian04 - in there interest of making our [commit history cleaner and easier to follow](https://github.com/cliffordfajardo/cato/blob/master/CONTRIBUTING.md#pull-requests), would you be able to do the following?

1. Make a separate pull request for the copy to clipboard utility? And inside this new pull request
   -  Modify the [`displayPotentialMathResult`](https://github.com/cliffordfajardo/cato/blob/master/app/util.js#L63) method to use the new utility
   - Modify the [`copy-url/index.js`](https://github.com/cliffordfajardo/cato/blob/master/app/plugins/copy-url/index.js) file to use the new utility

Also, we can remove the comment above the `event.preventDefault()` for the copy to clipboard utility, to make the code cleaner. MDN now has a section on what [event.preventDefault](https://developer.mozilla.org/en-US/docs/Web/Events/copy) does for the copy event 😄 

After these changes, I'll merge this to master. 

Next, for the shorten URL command in #9 we should also make a new pull request. Make sure you pull in all the recent changes from master. Now that the copyToClipboard utility will be available in master, in this new pull request you can make use of it 💪.

File: app/plugins/shorten-url/index.js [hunk 1]
```diff
@@ -0,0 +1,47 @@
+const browser = require('webextension-polyfill');
+const $ = require('jquery');
```
Comment (cliffordfajardo) [inline app/plugins/shorten-url/index.js]: We can probably do this without jQuery and use vanilla javascript so we don't add another dependency for making an AJAX request

File: app/plugins/shorten-url/index.js [hunk 2]
```diff
@@ -0,0 +1,42 @@
+const browser = require('webextension-polyfill');
+const {copyToClipboard} = require('../../util');
+const $ = require('jquery');
```
Comment (cliffordfajardo) [inline app/plugins/shorten-url/index.js:3]: We probably don't need to bring jQuery for this AJAX request. I haven't tested this, but something like this with async await. It makes our code more look and act synchronous. I haven't tested this out.

Also after the URL is copied, I create a new suggestion to display to the user, for which they can hit 'Enter' on. If we get an error back, we show them an error suggestion like our calculator command when you type `1+` for example.

Ex:

```js

const browser = require('webextension-polyfill');
const {copyToClipboard, renderSuggestions} = require('../../util');

const plugin = {
  keyword: "Shorten URL",
  subtitle: 'Copies a shortened version of the current url to your clipboard.',
  action: shortenUrl,
  icon: {
    path: 'images/shortened-url-icon'
  }
};


async function shortenUrl() {
  window.currentSearchSuggestions = []
  const tabs = await browser.tabs.query({active: true,currentWindow: true});
  const activeTabUrl = tabs[0].url;
  
  const request = new Request(postUrl, {
    'method': 'POST',
    'body': JSON.stringify({longUrl: activeTabUrl}),
    'mode': 'cors',
    'headers': new Headers({'Content-Type': 'application/json'})
  })
  
  try {
    const data = await (await fetch(request)).json()
    const suggestion = {
      keyword: `${data.id}`,
      subtitle: 'Copy shortened url to your clipboard.',
      action: copyToClipboard(data.id),
      icon: {
        path: 'images/shortened-url-icon'
      }
    }
  }
  catch (error) {
    const suggestion = {
      keyword: "Sorry, we couldn't shorten your URL at this time.",
      action: 'The network must be down or our API limit has execeded.',
      subtitle: 'Please enter a valid math expression.',
      icon: {
        path: 'images/shortened-url-icon'
      }
    }
  }
  finally {
    window.currentSearchSuggestions.push(suggestion)
    renderSuggestions(window.currentSearchSuggestions)
  }
}
```


For the icon references in the code above `'images/shortened-url-icon'` , you'll need to download [this icon](https://www.flaticon.com/free-icon/url_490604#term=url&page=1&position=3) and save it as an SVG file. Name it `shortened-url-icon` and then place it in the `app/images` folder to get it to show up.


I've been sick 😷  all week; apologies in advance if I my response is delayed.
I created this [gitter](https://gitter.im/cato-extension/) channel for chatting about Cato. We could use it to have more extended conversations about Cato, ask questions about code etc, without polluting the comments here

**Pull request closed without merging**